File: python/file_readers/read_csv_file.py
import csv

def read_csv_file(file_path: str):
    rows = []
    with open(file_path, 'r') as file:
        content = csv.reader(file)
        header = next(content)
        for row in content:
            rows.append(row)
    return {"header": header, "rows": rows}

File: javascript/fileReaders/readCsvFile.js
import fs from 'fs';
import csv from 'csv-parser';

const readCsvFile = (filePath) => {
  return new Promise((resolve, reject) => {
    const results = [];
    fs.createReadStream(filePath)
      .pipe(csv())
      .on('data', (row) => {
        results.push(row);
      })
      .on('end', () => {
        console.log('CSV file successfully processed');
        resolve(results);
      })
      .on('error', (err) => {
        console.error('An error occurred:', err);
        reject(err);
      });
  });
};

export { readCsvFile };
